elasticsearch-aws/src/main/scala/com/sumologic/elasticsearch/util/AwsRequestSigner.scala

```diff
@@ -171,7 +179,8 @@ class AwsRequestSigner(awsCredentialsProvider: AWSCredentialsProvider, region: S
   }
 
   private def hashedPayloadByteArray(httpRequest: HttpRequest): Array[Byte] = {
-    val data = httpRequest.entity.asString
+    val dataFuture = AkkaHttpUtil.entityToString(httpRequest.entity)
+    val data = Await.result(dataFuture, 1.second)
```

Comment:
what precisely is being waited for here? Are we waiting for the request to finish or is it just for some sort of string conversion?